Realm/RLMPredicateUtil.mm

```diff
@@ -72,8 +74,12 @@ - (NSExpression *)falseExpression;
         case NSMinusSetExpressionType:
             return [NSExpression expressionForMinusSet:visit(expression.leftExpression) with:visit(expression.rightExpression)];
 
-        case NSSubqueryExpressionType:
-            return [NSExpression expressionForSubquery:visit(expression.operand) usingIteratorVariable:expression.variable predicate:visit(expression.predicate)];
+        case NSSubqueryExpressionType: {
+            NSExpression *collection = expression.collection;
+            // NSExpression.collection is declared as id, but appears to always hold an NSExpression for subqueries.
+            REALM_ASSERT([collection isKindOfClass:[NSExpression class]]);
```

Comment:
`REALM_ASSERT_DEBUG()` should only be used if the check is too expensive to include in release builds.